### PR TITLE
Warn, Baby, Warn!

### PR DESCRIPTION
**Pull request by adamemerson**
Get rid of the Fun, New Compiler warnings from  GCC 9 and Clang 8.

Since the main cause are in software we include as submodules, specifically RocksDB and Google Test, just disable the relevant warning.

This does not fix the build failure of DPDK. I made an attempt to fix it but for some reason passing in flags to disable the warning made it unable to find its headers.

### CODE REVIEW

**Comment by adamemerson** [comment]
Jenkins, retest this please.

File: cmake/modules/BuildRocksDB.cmake [hunk 1]
```diff
@@ -44,6 +44,8 @@ function(build_rocksdb)
   if(HAS_WARNING_STRINGOP_TRUNCATION)
     list(APPEND rocksdb_CMAKE_ARGS -DCMAKE_C_FLAGS="-Wno-stringop-truncation")
   endif()
+  list(APPEND rocksdb_CMAKE_ARGS -DCMAKE_CXX_FLAGS="-Wno-deprecated-copy")
+
```
Comment (wjwithagen) [inline cmake/modules/BuildRocksDB.cmake]: @adamemerson 
My Clang builds also contain quite a few of the warnings below.
Perhaps a -Wno-shadow can suppress them?

```
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:111:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WAL_FILE_BYTES,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:190:3: note: previous declaration is here
  WAL_FILE_BYTES,   // Number of bytes written to WAL
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:112:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WAL_FILE_SYNCED,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:189:3: note: previous declaration is here
  WAL_FILE_SYNCED,  // Number of times WAL sync is done
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:113:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    BYTES_WRITTEN,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:126:3: note: previous declaration is here
  BYTES_WRITTEN,
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:114:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    NUMBER_KEYS_WRITTEN,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:119:3: note: previous declaration is here
  NUMBER_KEYS_WRITTEN,
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:115:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WRITE_DONE_BY_OTHER,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:195:3: note: previous declaration is here
  WRITE_DONE_BY_OTHER,  // Equivalent to writes done for others
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:116:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WRITE_DONE_BY_SELF,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:194:3: note: previous declaration is here
  WRITE_DONE_BY_SELF,
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:117:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WRITE_WITH_WAL,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:197:3: note: previous declaration is here
  WRITE_WITH_WAL,       // Number of Write calls that request WAL
  ^
7 warnings generated.
```

File: src/osd/PG.cc [hunk 2]
```diff
@@ -183,7 +183,7 @@ PG::PG(OSDService *o, OSDMapRef curmap,
   coll(p),
   osd(o),
   cct(o->cct),
-  pool(recovery_state.get_pool()),
+  pool(_pool),
```
Comment (tchaikov) [inline src/osd/PG.cc]: i'd say it's a bug fix.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
```
   -10> 2019-05-02 21:04:18.871 ffffa6673ad0  5 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) aio_write 0x500000~7ffff000 aio 0x2d147e10 (piece)
    -9> 2019-05-02 21:04:18.871 ffffa6673ad0  5 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) aio_write 0x804ff000~1000 aio 0x2d147cd0 (piece)
    -8> 2019-05-02 21:04:18.871 ffffa6673ad0 20 bluefs _flush_range caching tail of 0x400 and padding block with 0xc00
    -7> 2019-05-02 21:04:18.871 ffffa6673ad0 20 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) aio_write 0x80500000~3fffd000 (direct)
    -6> 2019-05-02 21:04:18.871 ffffa6673ad0 20 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) _aio_log_start 0x80500000~3fffd000
    -5> 2019-05-02 21:04:18.871 ffffa6673ad0  5 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) aio_write 0x80500000~3fffd000 aio 0xa591a010
    -4> 2019-05-02 21:04:18.871 ffffa6673ad0 20 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) aio_submit ioc 0x2d0f11d0 pending 3 running 0
    -3> 2019-05-02 21:04:30.131 ffffa539d5d0 20 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) _aio_log_finish 1 0x500000~7ffff000
    -2> 2019-05-02 21:04:30.131 ffffa539d5d0 -1 bdev(0x2d197000 ceph_test_bluefs.tmp.block.10329.5) aio to 5242880~2147479552 but returned: 2147418112
    -1> 2019-05-02 21:04:30.141 ffffa539d5d0 -1 /home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/os/bluestore/KernelDevice.cc: In function 'void KernelDevice::_aio_thread()' thread ffffa539d5d0 time 2019-05-02 21:04:30.145051
/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/os/bluestore/KernelDevice.cc: 543: ceph_abort_msg("unexpected aio return value: does not match length")

 ceph version Development (no_version) octopus (dev)
 1: (ceph::__ceph_abort(char const*, int, char const*, std::string const&)+0xe0) [0xffffa864c41c]
 2: (KernelDevice::_aio_thread()+0xc2c) [0x60b92c]
 3: (KernelDevice::AioCompletionThread::entry()+0x18) [0x613bc0]
 4: (Thread::entry_wrapper()+0x8c) [0xffffa85ede40]
 5: (Thread::_entry_func(void*)+0x14) [0xffffa85edda4]
 6: (()+0x7bb0) [0xffffb1ac7bb0]
 7: (()+0xdb4c0) [0xffffa6a6b4c0]
```

the failure on aarch64 is not related but weird enough.

File: cmake/modules/BuildRocksDB.cmake [hunk 3]
```diff
@@ -44,6 +44,8 @@ function(build_rocksdb)
   if(HAS_WARNING_STRINGOP_TRUNCATION)
     list(APPEND rocksdb_CMAKE_ARGS -DCMAKE_C_FLAGS="-Wno-stringop-truncation")
   endif()
+  list(APPEND rocksdb_CMAKE_ARGS -DCMAKE_CXX_FLAGS="-Wno-deprecated-copy")
```
Comment (tchaikov) [inline cmake/modules/BuildRocksDB.cmake]: probably we should enable this flag only if the compiler does support it.

**Comment by tchaikov** [comment]
@adamemerson ping?

File: cmake/modules/BuildRocksDB.cmake (hunk 3 above)
Comment (tchaikov) [reply cmake/modules/BuildRocksDB.cmake]: @adamemerson could you drop this commit? i think i managed to fix it in #28507.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -37,14 +37,12 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor")
...
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unknown-pragmas")
-CHECK_CXX_COMPILER_FLAG("-Wpessimizing-move" COMPILER_SUPPORTS_PESSIMIZING_MOVE)
-if(COMPILER_SUPPORTS_PESSIMIZING_MOVE)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wpessimizing-move")
-endif()
-CHECK_CXX_COMPILER_FLAG("-Wredundant-move" COMPILER_SUPPORTS_REDUNDANT_MOVE)
-if(COMPILER_SUPPORTS_REDUNDANT_MOVE)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wredundant-move")
-endif()
+foreach(cxx_flag "-Wpessimizing-move" "-Wredundant-move" "-Wno-deprecated-copy")
```
Comment (tchaikov) [inline src/CMakeLists.txt]: @adamemerson i reworked this part and dropped the change in its already included in #28507 . and more importantly, cmake make it a little bit quirky to pass flags to `CMAKE_ARGS` in `ExternalProject_Add()` as we need to be extra careful as cmake tries to be smart at escaping the spaces and quotes.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
retest this please

**Comment by tchaikov** [comment]
retest this please

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
weird...
```
175/176 Test   #4: smoke.sh ................................***Timeout 3600.06 sec
```

smoke.sh consistently timeouts. need to reproduce it locally.

**Comment by tchaikov** [comment]
retest this please

**Comment by tchaikov** [comment]
@adamemerson  i am willing to silence all the warnings! so i created #28684 containing your changes and incorporating my suggested change. so that i can include them in my rados qa run using the `build-integration-branch` script.

**Comment by adamemerson** [comment]
Sorry about that, I missed the notification for the change request.

**Pull request closed without merging**